browser/src/dom/NotebookbarAccessibility.js

```diff
@@ -229,21 +256,59 @@ var NotebookbarAccessibility = function() {
 			return; // Ignore shift key.
 		else if (key === 'ARROWUP') {
 			// Try to set focus on tab button.
-			if (this.activeTabPointers.id)
+			if (this.activeTabPointers.id) {
+				this.removeFocusFromTab();
 				document.getElementById(this.activeTabPointers.id).focus();
+			}
 		}
 		else if (key === 'ARROWDOWN') {
 			// Try to set focus on the first button of the tab content.
 			if (this.activeTabPointers.contentList.length > 0) {
 				for (var i = 0; i < this.activeTabPointers.contentList.length; i++) {
 					var element = document.getElementById(this.activeTabPointers.contentList[i].id);
 					if (element.tabIndex >= 0) {
+						this.removeFocusFromTab();
 						element.focus();
 						break;
 					}
 				}
 			}
 		}
+		else if (key === 'ARROWRIGHT') {
+			var currentSelectedTab = this.getCurrentSelectedTab();
+			var previousTab = currentSelectedTab.nextElementSibling;
+			if (!previousTab) {
+				previousTab = document.querySelector('.ui-tab.notebookbar:not(.hidden)');
+			}
+			while (previousTab) {
+			if (!previousTab.classList.contains('hidden')) {
+				// Found the next element without the "hidden" class
+				break;
+			}
+			previousTab = previousTab.nextElementSibling;
+			}
+			this.removeFocusFromTab();
+			previousTab.click();
+			previousTab.focus();
+		}
+		else if (key === 'ARROWLEFT') {
+			var currentSelectedTab = this.getCurrentSelectedTab();
```

Comment:
it would be good to create method getNextTab() to not duplicate code in ARROWLEFT / RIGHT

Reply:
thanks now looks good :)